xbmc/playlists/PlayListXSPF.cpp

```diff
@@ -0,0 +1,150 @@
+/*
+ *  Copyright (C) 2018 Tyler Szabo
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#include "PlayListXSPF.h"
+
+#include "utils/log.h"
+#include "utils/URIUtils.h"
+#include "utils/XBMCTinyXML.h"
+#include "URL.h"
+
+using namespace PLAYLIST;
+
+/*
+
+Specification: http://www.xspf.org/specs/
+
+Example:
+
+<?xml version="1.0" encoding="UTF-8"?>
+<playlist xmlns="http://xspf.org/ns/0/" version="1">
+  <title>KEXP</title>
+  <trackList>
+    <track>
+      <location>http://live-aacplus-64.kexp.org/kexp64.aac</location>
+      <title>KEXP.ORG 90.3FM - where the music matters</title>
+      <info>https://kexp.org/</info>
+      <image>https://www.kexp.org/static/assets/img/logo-white.svg</image>
+    </track>
+    <track><location>Relative/Path/To.mp3</location></track>
+    <track><location>file:///music/song_1.ogg</location></track>
+    <track><location>file:///music/song_2.flac</location></track>
+    <track><location>file:///music/song_3.mp3</location></track>
+  </trackList>
+</playlist>
+*/
+
+static constexpr char const * LOCATION_TAGNAME = "location";
+static constexpr char const * PLAYLIST_TAGNAME = "playlist";
+static constexpr char const * TITLE_TAGNAME = "title";
+static constexpr char const * TRACK_TAGNAME = "track";
+static constexpr char const * TRACKLIST_TAGNAME = "trackList";
+
+
+CPlayListXSPF::CPlayListXSPF(void) = default;
+
+CPlayListXSPF::~CPlayListXSPF(void) = default;
+
+
+bool CPlayListXSPF::Load(const std::string& strFileName)
+{
+  CXBMCTinyXML xmlDoc;
+
+  if (!xmlDoc.LoadFile(strFileName))
+  {
+    CLog::Log(LOGERROR, "Error parsing XML file %s (%d, %d): %s", strFileName.c_str(), xmlDoc.ErrorRow(), xmlDoc.ErrorCol(), xmlDoc.ErrorDesc());
+    return false;
+  }
+
+  TiXmlElement* pPlaylist = xmlDoc.FirstChildElement(PLAYLIST_TAGNAME);
+  if (!pPlaylist)
+  {
+    CLog::Log(LOGERROR, "Error parsing XML file %s: missing root element %s", strFileName.c_str(), PLAYLIST_TAGNAME);
+    return false;
+  }
+
+  TiXmlElement* pTracklist = pPlaylist->FirstChildElement(TRACKLIST_TAGNAME);
+  if (!pTracklist)
+  {
+    CLog::Log(LOGERROR, "Error parsing XML file %s (%d, %d): missing element %s", strFileName.c_str(), pPlaylist->Row(), pPlaylist->Column(), TRACKLIST_TAGNAME);
+    return false;
+  }
+
+  Clear();
+  URIUtils::GetParentPath(strFileName, m_strBasePath);
+
+  TiXmlElement* pPlaylistTitle = pPlaylist->FirstChildElement(TITLE_TAGNAME);
+  if (pPlaylistTitle)
+  {
+    m_strPlayListName = pPlaylistTitle->GetText();
+  }
+
+  TiXmlElement* pCurTrack = pTracklist->FirstChildElement(TRACK_TAGNAME);
+
+  while (pCurTrack)
+  {
+    std::string label = "";
+    std::string location = "";
+    std::string localpath = "";
```

Comment:
`= ""` isn't necessary for std::string, that's what the default constructor does anyway.